lib/gs-plugin-loader.c

```diff
@@ -4194,3 +4198,25 @@ gs_plugin_loader_get_locale (GsPluginLoader *plugin_loader)
 	GsPluginLoaderPrivate *priv = gs_plugin_loader_get_instance_private (plugin_loader);
 	return priv->locale;
 }
+
+gboolean
+gs_plugin_loader_app_copying (GsPluginLoader *plugin_loader, GsApp *app)
+{
+	GsPluginLoaderPrivate *priv = gs_plugin_loader_get_instance_private (plugin_loader);
+	if (app == NULL) {
+		return FALSE;
+	}
+	return GPOINTER_TO_UINT (g_hash_table_lookup (priv->global_plugin_statuses,
+				gs_app_get_id (app))) == GS_PLUGIN_STATUS_COPYING;
+}
+
+gboolean
+gs_plugin_loader_copy_queue_empty (GsPluginLoader *plugin_loader)
+{
+	GsPluginLoaderPrivate *priv = gs_plugin_loader_get_instance_private (plugin_loader);
+	GList *statuses = g_hash_table_get_values (priv->global_plugin_statuses);
```

Comment:
Should be `g_autoptr(GList)` since `g_hash_table_get_values` has its return value documented as `(transfer container)`